mantis-control-plane/mantis-control-plane-dynamodb/src/main/java/io/mantisrx/extensions/dynamodb/DynamoDBMasterMonitor.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright 2024 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.mantisrx.extensions.dynamodb;
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBLockClient;
+import com.amazonaws.services.dynamodbv2.LockItem;
+import io.mantisrx.server.core.BaseService;
+import io.mantisrx.server.core.json.DefaultObjectMapper;
+import io.mantisrx.server.core.master.MasterDescription;
+import io.mantisrx.server.core.master.MasterMonitor;
+import io.mantisrx.shaded.com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.time.Duration;
+import java.util.Optional;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import javax.annotation.Nullable;
+import lombok.extern.slf4j.Slf4j;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import rx.Observable;
+import rx.subjects.BehaviorSubject;
+
+@Slf4j
+public class DynamoDBMasterMonitor extends BaseService implements MasterMonitor {
```

Comment:
This design is a simple polling design to monitor the key in DynamoDB. After I finished this code, I know it can be vastly improved. I chose to not push further because there is potential to extract this. Leader monitoring could be completely separate from leader election. In the current implementation ZK and DynamoDB leader monitoring is very knowledgable about how leader is elected. If we change the design to the leader publishes when elected, and heartbeats regularly on a common messaging framework, then monitoring can be separated. This would push requirements to leader election, but I think we could actually push those to leadership manager. That refactor I thought should be a completely separate PR and this implementation should meet the current need.